src/databricks/labs/ucx/source_code/linters/files.py

```diff
@@ -193,60 +193,56 @@ def _process_dependency(
+        subdirectories are fixed.
+        """
         if path.is_dir():
+            fix_indicators = []
             for child_path in path.iterdir():
-                self.apply(child_path)
-            return True
-        return self._apply_file_fix(path)
+                fix_indicators.append(self.apply(child_path))
+            return all(fix_indicators)
+        return self._fix_file(path)
 
-    def _apply_file_fix(self, path: Path):
-        """
-        The fix method reads a file, lints it, applies fixes, and writes the fixed code back to the file.
+    def _fix_file(self, path: Path) -> bool:
+        """Fix a file.
+
+        Apply the code fixers on the read content of the file. If the fixers yield code changes, write them back to the
+        file.
         """
-        # Check if the file extension is in the list of supported extensions
-        if path.suffix not in self._extensions:
+        if not path.exists():
+            logger.warning(f"Skip non-existing file: {path}")
             return False
-        # Get the language corresponding to the file extension
-        language = self._extensions[path.suffix]
-        # If the language is not supported, return
+        language = self._extensions.get(path.suffix)
         if not language:
+            logger.warning(f"Skip fixing file with unsupported extension: {path}")
+            return False
+        logger.info(f"Fixing: {path}")
+        try:
+            code = path.read_text()
+        except UnicodeDecodeError as e:
+            logger.warning(f"Cannot decode file: {path}", exc_info=e)
             return False
-        logger.info(f"Analysing {path}")
-        # Get the linter for the language
         context = self._context_factory()
-        linter = context.linter(language)
-        # Open the file and read the code
-        with path.open("r") as f:
-            try:
-                code = f.read()
-            except UnicodeDecodeError as e:
-                logger.warning(f"Could not decode file {path}: {e}")
-                return False
-            applied = False
-            # Lint the code and apply fixes
-            for advice in linter.lint(code):
-                logger.info(f"Found: {advice}")
-                fixer = context.fixer(language, advice.code)
-                if not fixer:
-                    continue
-                logger.info(f"Applying fix for {advice}")
-                code = fixer.apply(code)
-                applied = True
-            if not applied:
-                return False
-            # Write the fixed code back to the file
-            with path.open("w") as f:
-                logger.info(f"Overwriting {path}")
-                f.write(code)
-                return True
+        if not context.is_supported(language):
+            logger.warning(f"Skip fixing unsupported language: {language.value}")
+            return False
+        fixed_code = context.apply_fixes(language, code)
```

Comment:
Rewrite method to use `apply_fixes` and be more consistent with other parts of the code base that read and write files